blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java

```diff
@@ -834,11 +838,17 @@ public String completeMultipartUpload(MultipartUpload mpu, List<MultipartPart> p
             throw propagate(ioe);
          }
          streams.add(is);
+         partHashes.append(blobPart.getMetadata().getETag());
       }
+      StringBuilder mpuETagBuilder = new StringBuilder("\"");
+      mpuETagBuilder.append(Hashing.md5().hashString(partHashes.toString(), US_ASCII).toString());
+      mpuETagBuilder.append(String.format("-%d\"", parts.size()));
+      String mpuETag = mpuETagBuilder.toString();
```

Comment:
Use the return values from `StringBuilder` and avoid the `String.format` call:

```java
String mpuETag = new StringBuilder("\"")
    .append(Hashing.md5().hashString(partHashes.toString(), US_ASCII).toString())
    .append("-")
    .append(parts.size())
    .toString();
```

Repeated above.

Reply:
Done